js/ui/codeView.js

```diff
@@ -160,46 +165,50 @@ var WindowTrackingButton = GObject.registerClass({
     Signals: {
         'clicked': {}
     },
-}, class WindowTrackingButton extends GObject.Object {
+}, class WindowTrackingButton extends St.Button {
     _init(params) {
         this._toolbox_window = null;
+        this._window = null;
+
+        let buttonParams = _getViewSourceButtonParams(true);
+        params = Params.parse(params, buttonParams, true);
```

Comment:
I think this could be achieved with `params = Object.assign({}, buttonParams, params);`? I know upstream is slowly trying to get rid of `params.js`, so that might be more future-proof.